### PR TITLE
perf: cache system cell for resolve deps

### PR DESCRIPTION
**Pull request by zhangsoledad**
The goal of this PR is to optimize transaction resolve.
Currently, transaction resolve is inefficient, cause system-cell is uncached, every transaction depends on system cell will read system-cell from disk repeatedly. this PR proposal a system-cell cache implementation.

### CODE REVIEW

**Comment by zhangsoledad** [comment]
benchmark

**Comment by yangby-cryptape** [comment]
#### Benchmark Result
  - TPS: 377.90
  - Samples Count: 50
  - CKB Version: e72ebc7ae24800b71cb9a295ac34d189fd3620ab
  - Instance Type: c5.xlarge
  - Instances Count: 3
  - Bench Type: 2in2out
  - CKB Logger Filter: info,ckb=debug

**Comment by doitian** [comment]
Please add PR description:

* Why this PR is needed
* What this PR has done
* The comparison

File: util/types/src/core/cell.rs [hunk 1]
```diff
@@ -424,16 +446,27 @@ pub fn resolve_transaction<CP: CellProvider, HC: HeaderChecker, S: BuildHasher>(
         }
     }
 
+    let system_cell = SYSTEM_CELL.get().expect("SYSTEM_CELL is not initialized");
```
Comment (doitian) [inline util/types/src/core/cell.rs:449]: I think a more elegant solution is creating a wrapper around the resolver, like:

```
struct<F> CellProviderWithSystemCellCache {
  system_cell: SystemCell,
  inner: F,
}
```

Comment (zhangsoledad) [reply util/types/src/core/cell.rs:449]: This PR is a draft for bench, I will improve it later.
But I think `CellProvider` is also a bad idea, this cache is for dep, not input.

**Comment by zhangsoledad** [comment]
benchmark

**Comment by zhangsoledad** [comment]
benchmark

**Comment by yangby-cryptape** [comment]
#### Benchmark Result
  - TPS: 381.56
  - Samples Count: 51
  - CKB Version: 0c86b9fb64b194d9bfad03e576b8468a54c35a49
  - Instance Type: c5.xlarge
  - Instances Count: 3
  - Bench Type: 2in2out
  - CKB Logger Filter: info,ckb=debug

**Pull request merged**